desc/backend.py

```diff
@@ -270,3 +271,29 @@ def while_loop(cond_fun, body_fun, init_val):
         while cond_fun(val):
             val = body_fun(val)
         return val
+
+    def vmap(fun, out_axes=0):
+        """A numpy implementation of jax.lax.map whose API is a subset of jax.vmap.
```

Comment:
I tested this by copying the same function to the if use_jax section and running the tests.

Reply:
Would `jnp.vectorize` work? We already use that for some other things I think. If not this is fine.

Reply:
I couldn't get `jnp.vectorize`  to work when the function changes dimension. The performance of vmap should be at least as fast as vectorize though.
```python
import jax.numpy as jnp
from jax import vmap

a = jnp.arange(6)
inputs = jnp.stack([a, a[::-1], -a])


def f(x):
    return x**3


def g(x):
    # downsample
    return f(x)[: x.size // 2]


def numpy_vmap(f, f_inputs):
    f_inputs = jnp.atleast_2d(f_inputs)
    f_outputs = jnp.stack([f(f_input) for f_input in f_inputs], axis=1)
    return jnp.atleast_1d(jnp.squeeze(f_outputs))


def jax_vmap(f, f_inputs):
    f_inputs = jnp.atleast_2d(f_inputs)
    f_outputs = vmap(f, out_axes=1)(f_inputs)
    return jnp.atleast_1d(jnp.squeeze(f_outputs))


print("1d input, vectorize works like vmap")
print(numpy_vmap(f, a))
print(jax_vmap(f, a))
print(jnp.vectorize(f)(a).T)
print("2d input, vectorize works like vmap")
print(numpy_vmap(f, inputs))
print(jax_vmap(f, inputs))
print(jnp.vectorize(f)(inputs).T)
print("vmap still works, but vectorize fails")
print(numpy_vmap(g, inputs))
print(jax_vmap(g, inputs))
print(jnp.vectorize(g)(inputs).T)
```